### PR TITLE
Remove `--disable-auto-batching` CLI option

### LINKED ISSUE #3294: Remove `--disable-auto-batching` CLI option
We should not keep the `--disable-auto-batching` CLI option, because:
1. I don't think there are clear advantages to disabling autobatching other than for debug purposes.
2. We can't introduce breaking changes after v1.0 is released. Keeping the flag will commit us to keeping this feature until v2.0 is released.
3. We can reintroduce it as an unstable feature later on.

## Impacted teams
@meilisearch/docs-team

### PR DESCRIPTION
**Pull request by loiclec**
Fixes #3294 

The `index-scheduler` code is not modified, only the CLI options have changed.

### CODE REVIEW

**Review by irevoire — APPROVED** [review]
@gmourier, you can bors merge if you want to be part of the contributors list 🎉

**Review by Kerollmops — APPROVED** [review]
It looks good to me, waiting for the approval of @gmourier 🐝

**Comment by gmourier** [comment]
bors merge

**Pull request merged**
